gcloud-java-dns/src/main/java/com/google/gcloud/dns/ManagedZoneInfo.java

```diff
@@ -0,0 +1,334 @@
+    public ManagedZoneInfo build() {
+      return new ManagedZoneInfo(this);
+    }
+  }
+
+  private ManagedZoneInfo(Builder builder) {
+    this.name = builder.name;
+    this.id = builder.id;
+    this.creationTimeMillis = builder.creationTimeMillis;
+    this.dnsName = builder.dnsName;
+    this.description = builder.description;
+    this.nameServerSet = builder.nameServerSet;
+    this.nameServers = ImmutableList.copyOf(builder.nameServers);
+  }
+
+  /**
+   * Returns a builder for {@code ManagedZoneInfo} with an assigned {@code name}.
+   */
+  public static Builder builder(String name) {
+    return new Builder(name);
+  }
+
+  /**
+   * Returns a builder for {@code ManagedZoneInfo} with an assigned {@code id}.
+   */
+  public static Builder builder(Long id) {
+    return new Builder(id);
+  }
+
+  /**
+   * Returns a builder for {@code ManagedZoneInfo} with an assigned {@code name} and {@code id}.
+   */
+  public static Builder builder(String name, Long id) {
+    return new Builder(name, id);
+  }
+
+  /**
+   * Returns an empty builder for {@code ManagedZoneInfo}. We use it internally in {@code toPb()}.
+   */
+  private static Builder builder() {
+    return new Builder();
+  }
+
+  /**
+   * Returns the user-defined name of the managed zone.
+   */
+  public String name() {
+    return name;
+  }
+
+  /**
+   * Returns the read-only managed zone id assigned by the server.
+   */
+  public Long id() {
+    return id;
+  }
+
+  /**
+   * Returns the time when this time that this managed zone was created on the server.
+   */
+  public Long creationTimeMillis() {
+    return creationTimeMillis;
+  }
+
+  /**
+   * Returns the DNS name of this managed zone, for instance "example.com.".
+   */
+  public String dnsName() {
+    return dnsName;
+  }
+
+  /**
+   * Returns the description of this managed zone. This is at most 1024 long mandatory string
+   * provided by the user.
+   */
+  public String description() {
+    return description;
+  }
+
+  /**
+   * Returns the optionally specified set of DNS name servers that all host this managed zone.
+   */
+  public String nameServerSet() {
+    // todo(mderka) update this doc after finding out more about this from the service owners
+    return nameServerSet;
+  }
+
+  /**
+   * The nameservers that the managed zone should be delegated to. This is defined by the Google DNS
+   * cloud.
+   */
+  public List<String> nameServers() {
+    return nameServers;
+  }
+
+  /**
+   * Returns a builder for {@code ManagedZoneInfo} prepopulated with the metadata of this managed
+   * zone.
+   */
+  public Builder toBuilder() {
+    return new Builder(this);
+  }
+
+  com.google.api.services.dns.model.ManagedZone toPb() {
+    com.google.api.services.dns.model.ManagedZone pb =
+        new com.google.api.services.dns.model.ManagedZone();
+    pb.setDescription(this.description());
+    pb.setDnsName(this.dnsName());
+    if (this.id() != null) {
+      pb.setId(BigInteger.valueOf(this.id()));
```

Comment:
If their ID can be more than Long.MAX_VALUE than I don't think you can use `long`.